controllers/usersController.js

```diff
@@ -0,0 +1,72 @@
+const db = require("../db/models");
+const User = db.users;
+
+const Op = db.Sequelize.Op;
+
+function createNewUser(req, res, next) {
+  const { email, password, role, firstname, lastname, gender, dob } = req.body;
+  User.create({
+    email,
+    password,
+    role,
+    firstname,
+    lastname,
+    gender,
+    dob,
+  })
+    .then((data) => {
+      res.status(201).send({ data });
+    })
+    .catch((err) => {
+      console.log(err);
+      next(err);
+    });
+}
+function verifyUser(req, res, next) {
+  const { token } = req.params;
+  const { email } = req.body;
+
+  User.findByPk(email)
+    .then((user) => {
+      if (user) {
+        if (user.isVerified) {
+          res.status(400).send({
+            message: "Account already verified!",
+          });
+        } else if (token !== "12345") {
+          res.status(400).send({
+            message: "Token is invalid",
+          });
+        } else {
+          User.update(
+            {
+              isVerified: true,
+            },
+            {
+              where: {
+                email: user.email,
+              },
+            }
+          )
+            .then((row) => {
+              res.status(200).send("Email verified! Please proceed to login");
+            })
+            .catch((err) => {
+              console.log(err);
+              next(err);
+            });
+        }
+      } else {
+        res.status(404).send({
+          message:
+            "Email is invalid. Please check that you've registered with the correct email",
```

Comment:
Nice work and good choose of the http response code. However, you can say Email not found in system or not registered instead of saying Email is invalid. The person might think the email is not a valid email in general.